### PR TITLE
Optimize get info 3.4.2

### PR DESCRIPTION
**Pull request by vakopian**

### CODE REVIEW

File: src/api/c/cast.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ using namespace detail;
 
 static af_array cast(const af_array in, const af_dtype type)
 {
-    const ArrayInfo& info = getInfo(in);
+    const ArrayInfo& info = getInfo(in, false, true);
```
Comment (pavanky) [inline src/api/c/cast.cpp]: Why has this changed?

Comment (umar456) [inline src/api/c/cast.cpp]: I think you are looking at an older commit

Comment (pavanky) [inline src/api/c/cast.cpp]: Not sure why github notified this to me as new..

Comment (vakopian) [inline src/api/c/cast.cpp]: I just merged from hotfix-3.4.2, which had that change. The github diff in the "files changed" tab actually doesn't show the diff like it is showing here. Must be github bug...

File: src/api/c/array.cpp [hunk 2]
```diff
@@ -368,7 +368,7 @@ af_err af_get_numdims(unsigned *nd, const af_array in)
     af_err fn1(bool *result, const af_array in)                 \
     {                                                           \
         try {                                                   \
-            ArrayInfo info = getInfo(in, false, false);         \
+            const ArrayInfo& info = getInfo(in, false, false);         \
```
Comment (umar456) [inline src/api/c/array.cpp]: Please align the line break character

**Comment by shehzan10** [comment]
build arrayfire linux ci
build arrayfire osx ci
build arrayfire windows ci

**Comment by shehzan10** [comment]
build arrayfire osx cuda ci
build arrayfire osx opencl ci

**Pull request merged**